### PR TITLE
adds 2 million lightswitches to donutstation

### LINKED ISSUE #43373: donutstation area issues;
Normally I'd resolve this myself but I'll be moving in under a week and it will likely be a bit before I can get to it.

Primary issues to focus on are non-maintenance areas with no air alarms & cameras, and areas with no APCs.
```
Test Areas

Only checking areas on station z-levels.


AREAS WITHOUT AN APC:
/area/maintenance/port/fore
/area/maintenance/port
/area/science/storage
/area/maintenance/port/aft
/area/construction/mining/aux_base
/area/maintenance/department/medical
/area/medical/medbay/lobby
/area/hallway/primary/central
/area/crew_quarters/kitchen/coldroom
/area/maintenance/aft
/area/maintenance/starboard
/area/maintenance/department/security
/area/bridge
/area/crew_quarters/heads/hos

AREAS WITHOUT AN AIR ALARM:
/area/maintenance/port/fore
/area/tcommsat/server
/area/maintenance/port
/area/science/research
/area/medical/virology
/area/science/storage
/area/science/server
/area/maintenance/port/aft
/area/medical/sleeper
/area/medical/medbay/central
/area/medical/cryo
/area/maintenance/department/medical
/area/maintenance/fore
/area/crew_quarters/dorms
/area/ai_monitored/storage/eva
/area/ai_monitored/turret_protected/aisat/service
/area/ai_monitored/turret_protected/aisat/foyer
/area/hallway/secondary/entry
/area/maintenance/aft
/area/quartermaster/qm
/area/maintenance/disposal
/area/maintenance/starboard
/area/maintenance/department/engine
/area/hydroponics/garden
/area/ai_monitored/nuke_storage
/area/lawoffice
/area/maintenance/starboard/fore
/area/security/brig
/area/maintenance/department/security
/area/security/main
/area/crew_quarters/heads/hos
/area/ai_monitored/security/armory
/area/security/prison

AREAS WITHOUT A REQUEST CONSOLE:
/area/maintenance/port/fore
/area/tcommsat/computer
/area/tcommsat/server
/area/maintenance/port
/area/science/research
/area/science/storage
/area/science/server
/area/medical/patients_rooms/room_a
/area/medical/patients_rooms/room_b
/area/maintenance/port/aft
/area/medical/sleeper
/area/construction/mining/aux_base
/area/crew_quarters/fitness
/area/medical/genetics
/area/maintenance/department/crew_quarters/bar
/area/medical/surgery
/area/medical/cryo
/area/science/explab
/area/construction
/area/crew_quarters/fitness/locker_room
/area/hallway/secondary/exit/departure_lounge
/area/maintenance/department/medical
/area/security/checkpoint/medical
/area/gateway
/area/maintenance/fore
/area/crew_quarters/fitness/recreation
/area/crew_quarters/dorms
/area/science/robotics/mechbay
/area/crew_quarters/theatre
/area/hallway/primary/central
/area/engine/gravity_generator
/area/library
/area/ai_monitored/storage/eva
/area/library/lounge
/area/crew_quarters/kitchen/coldroom
/area/maintenance/disposal/incinerator
/area/ai_monitored/turret_protected/aisat/service
/area/security/checkpoint
/area/ai_monitored/turret_protected/aisat/foyer
/area/hallway/secondary/entry
/area/engine/engineering
/area/engine/engine_room
/area/maintenance/aft
/area/engine/engine_room/external
/area/quartermaster/office
/area/vacant_room/commissary
/area/chapel/main
/area/maintenance/disposal
/area/quartermaster/warehouse
/area/maintenance/starboard
/area/maintenance/department/engine
/area/teleporter
/area/hydroponics/garden
/area/ai_monitored/nuke_storage
/area/security/courtroom
/area/storage/tech
/area/maintenance/starboard/fore
/area/security/brig
/area/medical/morgue
/area/maintenance/department/security
/area/security/detectives_office
/area/ai_monitored/turret_protected/ai_upload
/area/security/main
/area/crew_quarters/heads/captain/private
/area/ai_monitored/security/armory
/area/security/prison
/area/security/processing

AREAS WITHOUT ANY LIGHTS:
/area/maintenance/department/crew_quarters/bar
/area/maintenance/department/medical
/area/security/checkpoint/medical
/area/crew_quarters/kitchen/coldroom
/area/ai_monitored/turret_protected/aisat/service
/area/maintenance/department/security

AREAS WITHOUT A LIGHT SWITCH:
/area/maintenance/port/fore
/area/tcommsat/computer
/area/tcommsat/server
/area/maintenance/port
/area/science/research
/area/science/storage
/area/science/mixing
/area/science/server
/area/maintenance/port/aft
/area/crew_quarters/heads/hor
/area/medical/sleeper
/area/construction/mining/aux_base
/area/medical/medbay/central
/area/crew_quarters/fitness
/area/medical/genetics
/area/medical/genetics/cloning
/area/medical/surgery
/area/medical/cryo
/area/construction
/area/crew_quarters/fitness/locker_room
/area/hallway/secondary/exit/departure_lounge
/area/security/checkpoint/science
/area/maintenance/department/medical
/area/medical/storage
/area/security/checkpoint/medical
/area/maintenance/fore
/area/crew_quarters/fitness/recreation
/area/medical/chemistry
/area/medical/medbay/lobby
/area/science/robotics/mechbay
/area/crew_quarters/theatre
/area/crew_quarters/bar
/area/hallway/primary/central
/area/engine/gravity_generator
/area/library
/area/hydroponics
/area/ai_monitored/storage/eva
/area/library/lounge
/area/crew_quarters/kitchen/coldroom
/area/ai_monitored/turret_protected/aisat/service
/area/ai_monitored/turret_protected/ai
/area/security/checkpoint
/area/ai_monitored/turret_protected/aisat/foyer
/area/hallway/secondary/entry
/area/engine/engineering
/area/engine/engine_room
/area/maintenance/aft
/area/engine/engine_room/external
/area/quartermaster/office
/area/quartermaster/storage
/area/janitor
/area/chapel/main
/area/maintenance/disposal
/area/quartermaster/miningdock
/area/maintenance/starboard
/area/maintenance/department/engine
/area/teleporter
/area/hydroponics/garden
/area/ai_monitored/nuke_storage
/area/storage/tech
/area/maintenance/starboard/fore
/area/security/brig
/area/medical/morgue
/area/maintenance/department/security
/area/bridge
/area/ai_monitored/turret_protected/ai_upload
/area/security/main
/area/crew_quarters/heads/hos
/area/crew_quarters/heads/captain/private
/area/crew_quarters/heads/captain
/area/ai_monitored/security/armory
/area/security/prison
/area/security/processing

AREAS WITHOUT ANY INTERCOMS:
/area/maintenance/port/fore
/area/science/xenobiology
/area/tcommsat/computer
/area/tcommsat/server
/area/maintenance/port
/area/science/research
/area/medical/virology
/area/science/storage
/area/science/mixing
/area/science/server
/area/medical/patients_rooms/room_a
/area/medical/patients_rooms/room_b
/area/maintenance/port/aft
/area/crew_quarters/heads/hor
/area/medical/sleeper
/area/construction/mining/aux_base
/area/medical/medbay/central
/area/crew_quarters/fitness
/area/medical/genetics
/area/medical/genetics/cloning
/area/maintenance/department/crew_quarters/bar
/area/medical/surgery
/area/medical/cryo
/area/science/explab
/area/construction
/area/crew_quarters/fitness/locker_room
/area/hallway/secondary/exit/departure_lounge
/area/crew_quarters/heads/cmo
/area/security/checkpoint/science
/area/maintenance/department/medical
/area/medical/storage
/area/security/checkpoint/medical
/area/science/robotics/lab
/area/gateway
/area/maintenance/fore
/area/science/lab
/area/crew_quarters/fitness/recreation
/area/medical/chemistry
/area/medical/medbay/lobby
/area/crew_quarters/dorms
/area/science/robotics/mechbay
/area/crew_quarters/theatre
/area/crew_quarters/bar
/area/hallway/primary/central
/area/engine/gravity_generator
/area/library
/area/hydroponics
/area/ai_monitored/storage/eva
/area/library/lounge
/area/crew_quarters/kitchen/coldroom
/area/crew_quarters/kitchen
/area/maintenance/disposal/incinerator
/area/ai_monitored/turret_protected/aisat/service
/area/ai_monitored/turret_protected/ai
/area/security/checkpoint
/area/ai_monitored/turret_protected/aisat/foyer
/area/hallway/secondary/entry
/area/engine/atmos
/area/engine/engineering
/area/engine/engine_room
/area/maintenance/aft
/area/engine/engine_room/external
/area/quartermaster/office
/area/vacant_room/commissary
/area/quartermaster/storage
/area/quartermaster/qm
/area/janitor
/area/chapel/main
/area/maintenance/disposal
/area/quartermaster/miningdock
/area/quartermaster/warehouse
/area/maintenance/starboard
/area/maintenance/department/engine
/area/crew_quarters/heads/chief
/area/teleporter
/area/hydroponics/garden
/area/ai_monitored/nuke_storage
/area/security/courtroom
/area/chapel/office
/area/lawoffice
/area/storage/tech
/area/storage/tools
/area/maintenance/starboard/fore
/area/security/brig
/area/medical/morgue
/area/maintenance/department/security
/area/security/detectives_office
/area/crew_quarters/heads/hop
/area/bridge
/area/ai_monitored/turret_protected/ai_upload
/area/security/main
/area/crew_quarters/heads/hos
/area/crew_quarters/heads/captain/private
/area/crew_quarters/heads/captain
/area/security/warden
/area/ai_monitored/security/armory
/area/security/prison
/area/security/processing

AREAS WITHOUT ANY CAMERAS:
/area/maintenance/port/fore
/area/maintenance/port
/area/medical/patients_rooms/room_a
/area/medical/patients_rooms/room_b
/area/maintenance/port/aft
/area/construction/mining/aux_base
/area/maintenance/department/crew_quarters/bar
/area/construction
/area/crew_quarters/fitness/locker_room
/area/security/checkpoint/science
/area/maintenance/department/medical
/area/security/checkpoint/medical
/area/maintenance/fore
/area/crew_quarters/dorms
/area/crew_quarters/kitchen/coldroom
/area/maintenance/aft
/area/maintenance/disposal
/area/quartermaster/miningdock
/area/quartermaster/warehouse
/area/maintenance/department/engine
/area/maintenance/starboard/fore
/area/maintenance/department/security
```

### PR DESCRIPTION
**Pull request by MMMiracles**
## About The Pull Request

Fixes #43373 

- Fixes missing APCs
- Fixes missing air alarms (how did nobody notice either of these FUCK)
- Adds a few extra request consoles (mostly for places like genetics and the main security equipment room).
- Adds a bunch of intercoms because I apparently forgot these exist
- Adds a few lights in missing areas
- added like 1 whole light switch

## Changelog
tweak: Fixed missing components on Donutstation.
/:cl:

the list looks bad until you realize over half of it it because a room doesn't have a switch/intercom/request console i swear it isn't that bad

### CODE REVIEW

**Comment by optimumtact** [comment]
shedding some light on donut's bad mapping

**Comment by MMMiracles** [comment]
i got this real nice fruit juicer for last christmas and i haven't had the opportunity to use it yet

**Comment by FloranOtten** [comment]
Have you traded hands with the Robot Devil or are you just that good at speedmapping

**Comment by ShizCalev** [comment]
Conflict.

**Comment by SpaceManiac** [comment]
Did what I could to resolve the conflict, please pull and then double-check

**Comment by optimumtact** [comment]
why?

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 1]
```diff
@@ -29937,6 +30113,13 @@
 /obj/machinery/light{
 	dir = 8
 	},
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: intentional?

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 2]
```diff
@@ -29937,6 +30113,13 @@
 /obj/machinery/light{
 	dir = 8
 	},
+/obj/item/radio/intercom{
+	broadcasting = 1;
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 3]
```diff
@@ -53746,21 +54086,13 @@
...
 /turf/open/floor/plasteel/showroomfloor,
 /area/crew_quarters/fitness/locker_room)
-"jsm" = (
-/obj/structure/lattice/catwalk,
-/obj/machinery/atmospherics/pipe/heat_exchanging/simple{
-	dir = 4
+"jtr" = (
+/obj/structure/window/reinforced{
+	dir = 1;
+	pixel_y = 1
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 4]
```diff
@@ -4743,6 +4779,13 @@
 	name = "HoochMaster Deluxe";
 	pixel_x = -4
 	},
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: Got a couple more of these still

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 5]
```diff
@@ -13749,6 +13821,13 @@
 /area/security/brig)
 "aFt" = (
 /obj/structure/closet/secure_closet/freezer/cream_pie,
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 6]
```diff
@@ -20482,7 +20616,7 @@
 	broadcasting = 1;
 	frequency = 1485;
 	listening = 0;
-	name = "Station Intercom (Medbay)";
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 7]
```diff
@@ -34194,6 +34375,13 @@
 	pixel_y = 7
 	},
 /obj/item/pen,
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 8]
```diff
@@ -53346,18 +53719,17 @@
...
-	},
-/obj/machinery/atmospherics/pipe/simple/supply/hidden{
-	dir = 4
+"hou" = (
+/obj/machinery/recharge_station,
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 9]
```diff
@@ -53805,54 +54132,59 @@
...
-/turf/open/floor/plasteel,
-/area/engine/engineering)
-"jVL" = (
+"jRN" = (
+/obj/structure/table/wood,
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

File: _maps/map_files/Donutstation/Donutstation.dmm [hunk 10]
```diff
@@ -54538,47 +55135,51 @@
...
-"ooZ" = (
-/obj/machinery/door/airlock/engineering/glass{
-	name = "Laser Room";
-	req_access_txt = "10"
+"opq" = (
+/obj/item/radio/intercom{
+	broadcasting = 1;
+	frequency = 1485;
+	listening = 0;
+	name = "station intercom";
```
Comment (ShizCalev) [inline _maps/map_files/Donutstation/Donutstation.dmm]: .

**Comment by AnturK** [comment]
Sorry for the conflicts, didn't notice this was ready.

**Pull request merged**